src/extlibs/hrvo/hrvo_agent.cpp

```diff
@@ -75,38 +82,69 @@ void HRVOAgent::updatePrimitive(const TbotsProto::Primitive &new_primitive,
         // multiple path points and remove this check
         CHECK(motion_control.path().points().size() >= 2)
             << "Empty path: " << motion_control.path().points().size() << std::endl;
-        auto destination = motion_control.path().points().at(1);
+        Point destination = createPoint(motion_control.path().points().at(1));
 
-        // Max distance which the robot can travel in one time step + scaling
-        // TODO (#2370): This constant is calculated multiple times.
-        float path_radius = (max_speed_ * simulator_->getTimeStep()) / 2;
-        auto path_points  = {PathPoint(
-            Vector(destination.x_meters(), destination.y_meters()), speed_at_dest)};
-        path              = AgentPath(path_points, path_radius);
-
-        // Update static obstacles
+        // Update non-robot obstacles
         std::set<TbotsProto::MotionConstraint> motion_constraints;
+        auto motion_constraint_enum_descriptor =
+            TbotsProto::MotionConstraint_descriptor();
         for (int constraint_int : motion_control.motion_constraints())
         {
-            if (!TbotsProto::MotionConstraint_IsValid(constraint_int))
+            TbotsProto::MotionConstraint constraint;
+            auto enum_value_descriptor =
+                motion_constraint_enum_descriptor->value(constraint_int);
+            bool parsed = TbotsProto::MotionConstraint_Parse(
+                enum_value_descriptor->name(), &constraint);
+            if (!parsed)
             {
+                LOG(WARNING) << "HRVO: Failed to parse MotionConstraint enum value: "
+                             << enum_value_descriptor->name();
                 continue;
             }
 
-            const auto constraint =
-                static_cast<TbotsProto::MotionConstraint>(constraint_int);
             auto new_obstacles =
                 obstacle_factory.createFromMotionConstraint(constraint, world);
-            if (constraint == TbotsProto::MotionConstraint::HALF_METER_AROUND_BALL)
+            if (enum_value_descriptor->options().HasExtension(TbotsProto::dynamic) &&
+                enum_value_descriptor->options().GetExtension(TbotsProto::dynamic))
             {
-                ball_obstacle = new_obstacles[0];
+                dynamic_obstacles.insert(dynamic_obstacles.end(), new_obstacles.begin(),
+                                         new_obstacles.end());
+                break;
```

Comment:
why don't we look at the other motion constraints if we see a dynamic one?

Reply:
Good point, I think I added the `break` by accident